src/neo4j_graphrag/experimental/pipeline/config/parser.py

```diff
@@ -0,0 +1,199 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import importlib
+from pathlib import Path
+from typing import Any, Optional, Union
+
+import neo4j
+
+from neo4j_graphrag.experimental.pipeline.config.param_resolvers import PARAM_RESOLVERS
+from neo4j_graphrag.experimental.pipeline.config.reader import ConfigReader
+from neo4j_graphrag.experimental.pipeline.config.types import (
+    ClassConfig,
+    DriverConfig,
+    ParamConfig,
+    ParamToResolveConfig,
+    SimpleKGPipelineConfig,
+)
+from neo4j_graphrag.experimental.pipeline.kg_builder import (
+    SimpleKGPipeline,
+    SimpleKGPipelineModel,
+)
+
+
+class SimpleKGPipelineConfigParser:
+    """Parses a config object into the required objects
+    to instantiate a SimpleKGPipeline. For instance, config
+    contains LLM information in this format:
+
+    .. code-block:: json
+
+       {
+        "llm_config": {
+            "name_": "openai",
+            "class_": "OpenAILLM",
+            "params_": {
+                "api_key": {
+                    "resolver_": "ENV",
+                    "var_": "OPENAI_API_KEY"
+                },
+                "model_name": "gpt-4o"
+            }
+       }
+
+    That must be parsed into an OpenAILLM instance with parameters
+    `model_name` and resolved `api_key`.
+
+    Args:
+        config (SimpleKGPipelineConfig): An instance of the SimpleKGPipelineConfig.
+    """
+
+    def __init__(self, config: SimpleKGPipelineConfig) -> None:
+        self.config = config
+
+    def _get_class(self, klass_path: str, optional_module: Optional[str] = None) -> Any:
```

Comment:
```suggestion
    def _get_class(self, class_path: str, optional_module: Optional[str] = None) -> Any:
```

And the same for klass below

Reply:
I've updated all the 'klass_*` names except the `klass` one because `class` is a reserved keyword.